### DISCUSSION TITLE
Electrical LUT_enforced

### DISCUSSION THREAD

**Discussion by SebGue** [Eomys/pyleecan#576] [Q&A]
Hello @BonneelP 

I hope you can answer my question without affort, ie. studying the code :-)

Now, in ElecLUTdq.run() if there is no enforced LUT a new LUT is calculated. Then the calculated LUT should be set as LUT_enforced (line 89 - `output.simu.elec.LUT_enforced = LUT`).
Contrary, e.g. in "tuto_Efficiency_map.ipynb" of PR #558, elec.LUT_enforced is still None after the simulation (without additional affort).
Is this behavior caused by VarSimu due to copying the simulation? But get the var. simulations copied only after the reference simulation? Otherwise each variation would run LUT simulation.

Best regards, Sebastian

**Comment by BonneelP** [comment]
Hello, 
I'm not sure if I will answer correctly but here are some hints that may help:
For me a key principal is that run should not modify the simulation otherwise running a simulation twice in a row can give different results (we had the issue with import_file for MagFEMM). But in the particular case of VarSimu, the idea of running a reference first is to reuse some data cf VarSimu.set_reused_data. So I guess we copy the ref simu before running but with is_reuse_LUT=True (or something like that) the N simulation are altered to use the LUT computed by the reference.
Hope it helps,
Best regards,
Pierre

**Reply by SebGue** [reply]
Hello Pierre,

thank you for your prompt responses! You were right with 'is_reuse_LUT'.

I know the principle 'don't modify the input' and that's what confused me, i.e. the line 89 assignment of LUT to simulation. On the other hand it would be really bad to (re)calculate the LUT each operation point. Nevertheless this contradicts the principle in a more or less clear manner, i.e. for single simulations at the top level 'simu' and for multi simulations hidden in output_list[x].simu and xoutput_ref.simu (if all output is keept).

I don't know if there is a better way to do this or if this is even an issue? 
Anyway thanks again for you helping me to understand the 'internal mechanics'.

Best regards, Sebastian

**Comment by faraz4714** [comment]
Dear all,
Thank you for your help in advance,
I saw the tutorial for efficiency mapping, but I have a question if I want to compute the efficiency for a specific operating point( I mean for only one condition, what should I do?
Best regards,
Faraz Mirali
